internal/utils/store.go

```diff
@@ -73,7 +73,7 @@ func GetHostMountPathFromSecretRef(ctx context.Context, client client.Client, lo
 
 // StorageProviderFromInfraProvider converts infra to object store provider.
 func StorageProviderFromInfraProvider(infra *druidv1alpha1.StorageProvider) (string, error) {
-	if infra == nil || len(*infra) == 0 {
+	if infra == nil {
```

Comment:
I understand this is renamed, but I was thinking for all cloudProvider semantics, shouldn't this be in a dedicated package like `store` and not as `utils`. 
This is easy to then expand if we bring in anything related to provider into the semantics of Druid. 
Also give a pattern for the external contributor to know where to keep cloudProvider specific semantics.

Reply:
Hmm this makes sense. Will make the change.

Reply:
Addressed in https://github.com/gardener/etcd-druid/pull/777/commits/a13614e155c87c8f7b7cd8269a9d1f9a023c1c04.

@ashwani2k PTAL and resolve.

Reply:
Thanks looks good.